tests/contenttypes_tests/test_checks.py

```diff
@@ -0,0 +1,247 @@
+            content_type = models.IntegerField()  # should be ForeignKey
+            object_id = models.PositiveIntegerField()
+            content_object = GenericForeignKey(
+                'content_type', 'object_id')
+
+        errors = Model.content_object.check()
+        expected = [
+            checks.Error(
+                "'Model.content_type' is not a ForeignKey.",
+                hint=(
+                    "GenericForeignKeys must use a ForeignKey to "
+                    "'contenttypes.ContentType' as the 'content_type' field."
+                ),
+                obj=Model.content_object,
+                id='contenttypes.E003',
+            )
+        ]
+        self.assertEqual(errors, expected)
+
+    def test_content_type_field_pointing_to_wrong_model(self):
+        class Model(models.Model):
+            content_type = models.ForeignKey('self', models.CASCADE)  # should point to ContentType
+            object_id = models.PositiveIntegerField()
+            content_object = GenericForeignKey(
+                'content_type', 'object_id')
+
+        errors = Model.content_object.check()
+        expected = [
+            checks.Error(
+                "'Model.content_type' is not a ForeignKey to 'contenttypes.ContentType'.",
+                hint=(
+                    "GenericForeignKeys must use a ForeignKey to "
+                    "'contenttypes.ContentType' as the 'content_type' field."
+                ),
+                obj=Model.content_object,
+                id='contenttypes.E004',
+            )
+        ]
+        self.assertEqual(errors, expected)
+
+    def test_missing_object_id_field(self):
+        class TaggedItem(models.Model):
+            content_type = models.ForeignKey(ContentType, models.CASCADE)
+            # missing object_id field
+            content_object = GenericForeignKey()
+
+        errors = TaggedItem.content_object.check()
+        expected = [
+            checks.Error(
+                "The GenericForeignKey object ID references the non-existent field 'object_id'.",
+                obj=TaggedItem.content_object,
+                id='contenttypes.E001',
+            )
+        ]
+        self.assertEqual(errors, expected)
+
+    def test_field_name_ending_with_underscore(self):
+        class Model(models.Model):
+            content_type = models.ForeignKey(ContentType, models.CASCADE)
+            object_id = models.PositiveIntegerField()
+            content_object_ = GenericForeignKey(
+                'content_type', 'object_id')
+
+        errors = Model.content_object_.check()
+        expected = [
+            checks.Error(
+                'Field names must not end with an underscore.',
+                obj=Model.content_object_,
+                id='fields.E001',
+            )
+        ]
+        self.assertEqual(errors, expected)
+
+    @override_settings(INSTALLED_APPS=['django.contrib.auth', 'django.contrib.contenttypes', 'contenttypes_tests'])
```

Comment:
I see, I suppose this check should be skipped when the `auth` app is not available. In the mean time feel free to ignore this comment.